src/python/pants/backend/python/dependency_inference/default_module_mapping.py

```diff
@@ -1,28 +1,23 @@
 # Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-# notice: using sets here to ensure the mapping is hashable
+# NB: You must use lowercase and replace all `_` and `.` with `-` for the requirement's name.
```

Comment:
... replace all runs of ... with a single `-` : maybe just use less words and force people to read the authoritative spec.